ngx_http_secure_token_processor_base.c

```diff
@@ -218,6 +212,14 @@ ngx_http_secure_token_url_state_machine(
 		switch (ctx->state)
 		{
 		case STATE_URL_SCHEME:
+			if (ch == ':')
```

Comment:
small optimization - please put the condition inside the next if block (`if (ch == scheme_delimeter[ctx->scheme_delim_pos])`). 
and (style) - combine the two conditions with && - `if (ch == ':' && !ngx_http_secure_token_is_http_scheme(ctx))`